src/transformers/generation_tf_utils.py

```diff
@@ -911,7 +1397,45 @@ def _generate_beam_search(
             assert (len(hypo) == max_length for hypo in best)
             decoded = tf.stack(best)
 
-        return decoded
+        if return_dict_in_generate:
+            if do_sample:
```

Comment:
those are some very nested if-statements -> could we maybe change it slighly to:

```if do_sample and self.config.is_encoder_decoder:
   ...
   elif do_sample and not self.config.is_encoder_decoder:
   ....
```

Reply:
Agree, I didn't like these nested if-statements either. :)